packages/core/src/search/search.ts

```diff
@@ -20,6 +20,7 @@ export interface SearchRequest<T extends Resource = Resource> {
   summary?: 'true' | 'text' | 'data';
   format?: string;
   pretty?: boolean;
+  types?: T['resourceType'][];
```

Comment:
This all looks good to me, but should we try to move towards this as the source of truth property and deprecate `resourceType`?

Reply:
> should we try to move towards this as the source of truth property and deprecate resourceType?

That's exactly the right question.

I'm open to feedback, but my thinking is "yes, but not right now"

Considerations:
* Overall API design
    * Currently a bunch of the search methods use TypeScript magic to map `resourceType` to the return type.
    * Both in `MedplumClient` and `Repository`, which are both used all over the place.
    * There is probably a way to do that with an array field, but it might be pretty gross.  
    * I'd like to think about it in the context of `_include` and `_revinclude` which also breaks the current abstraction
    * I'd like to think about it in the context of R5, which would be a different return type for the same `resourceType`
* Migration cost
    * That would be a breaking API change at a pretty critical depth.  It would probably necessitate a major version bump.  
    * Furthermore, internally we have a ton of code using the simple case of singular `resourceType`.  It would be a straightforward migration, but it would be a lot of wood to chop.

For better or worse, for right now, the common case is 1:1 single resource type input, single resource type output.  I think it makes sense to optimize for the common case.

I think the future state is probably something like this:

```ts
interface BaseSearchRequest<T extends BaseResource> { ... }

interface SingleTypeSearchRequest<T extends BaseResource> extends BaseSearchRequest<T> { ... };

interface MultiTypeSearchRequest<T extends BaseResource> extends BaseSearchRequest<T> { ... };

type SearchRequest<T extends BaseResource> = SingleTypeSearchRequest<T> | MultiTypeSearchRequest<T>;

type R4SearchRequest = SearchRequest<R4BaseResource>;

type R5SearchRequest = SearchRequest<R5BaseResource>;

// etc
```

So, back to your original question:  Yes, I think we should do that eventually, but I fear that it is a pretty big can of worms, and we should be ready for an onslaught of work to implement it soup to nuts.  By adding a single optional override member, we unblock the `_type` feature in a pretty surgical way.

tl;dr - I think this is acceptable tech debt.